source/develop/release-management/features/network/openstack_identity_api_v3.html.md

```diff
@@ -0,0 +1,290 @@
+      "methods" : [ "password" ],
+      "password" : {
+        "user" : {
+          "domain" : {
+            "name" : "Default"
+          },
+          "name" : "admin",
+          "password" : "c314d360aab94a5f"
+        }
+      }
+    },
+    "scope" : {
+      "project" : {
+        "domain" : {
+          "name" : "Default"
+        },
+        "name" : "admin"
+      }
+    }
+  }
+}
+~~~~
+The `scope` might be not required, e.g. if a default project is configured for
+the user. The relevant part of the response looks like this:
+~~~~
+HTTP/1.1 201 Created
+X-Subject-Token: gAAAAABbH3sSGLgtxvoHyYDLZd6fSuserBHG2P-3TdykaYnq-eYo7aUBdZK0I
+Vary: X-Auth-Token
+Content-Type: application/json
+
+{"token": ... }
+~~~~
+
+### Implementation in oVirt Engine
+
+To support OpenStack Identity API v3 oVirt engine has to be extended to handle
+the user domain name, the project name and the domain name of the project.
+This requires changes in webadmin, REST-API, the database and the 
+ExternalNetworkTokenProvider in the backend.
+
+The version of the OpenStack Identity API is derived from the authentication
+URL. If the authentication URL ends with "/v3" or "/v3/", OpenStack Identity
+API v3 will be used, else OpenStack Identity API v2.0 to ensure backward
+compatibility.
+
+The default ovirt-provider-ovn has not be modified, because the OpenStack
+Identity API v2.0 is not removed.
+
+### Implementation in oVirt Provider OVN
+To support OpenStack Identity API v3 the oVirt Provider OVN to be extended to
+handle requests to `/v3/auth/tokens`. The `scope` can be ignored, like the
+`tenantName`.
+In oVirt Provider OVN's OpenStack Identity API v2.0 implementation the
+user name has to match the following format:
+`<admin_username>[@<fqdn>]@<ovirt_profile>`, while the optional `<fqdn>` is
+the Active Directory or LDAP domain. For the reason of consistency, this format
+is kept in the OpenStack Identity API v3 implementation and the `user.domain`
+is ignored.
+
+## Prerequisites
+
+oVirt Engine requires an updated version of [OpenStack Java SDK 3.2.x][9] to
+accessing OpenStack Identity API v3.
+
+## Limitations
+
+OpenStack Identity API provides a catalog of multiple endpoints, which
+provides the URL to access the OpenStack service. This feature does change
+the current implementation in oVirt Engine, which requires the user to specify
+this URL manually.
+
+## Benefit to oVirt
+
+Support for OpenStack Identity API v3 in oVirt Engine is required to enable 
+access to services from [OpenStack instances which does not support Identity
+API v2.0, especially the OpenStack releases since Qeens.][12]
+
+## User Experience
+
+The "Add Provider" popup has to provide inputs for the new properties user
+domain name, the project name and the domain name of the project.
+If the authentication URL ends with "/v3" or "/v3/", the user domain name, the
+project name and the domain name of the project will be used, else the tenant
+name will be used.
+
```

Comment:
Hi
Project name, user domain and and project domain are new authentication for v3 ?
Does engine-setup will configure them when choosing to install ovn-provider? or does it should be set manually? does v3 now will be set on new engine-setup + ovn installation?
Please explain

Reply:
I added a new paragraph about engine-setup. You are welcome to discuss if you think I miss something.